includes/class-amp-theme-support.php

```diff
@@ -54,6 +132,44 @@ public static function register_hooks() {
 		// @todo Add character conversion.
 	}
 
+	/**
+	 * Prepends template hierarchy with template_path for AMP paired mode templates.
+	 *
+	 * @see get_query_template()
+	 *
+	 * @param array $templates Template hierarchy.
+	 * @returns array Templates.
+	 */
+	public static function filter_paired_template_hierarchy( $templates ) {
+		$support = get_theme_support( 'amp' );
+		$args    = array_shift( $support );
+		if ( isset( $args['template_path'] ) ) {
+			$amp_templates = array();
+			foreach ( $templates as $template ) {
+				$amp_templates[] = $args['template_path'] . '/' . $template;
```

Comment:
Since `template_path` is a _relative_ path from the theme root to locate the AMP templates directory, is this still the best name?

Reply:
I've updated these to use `template_dir` which is what `AMP_Post_Template` is calling it, too.  Setting this will look like the following, which should be easier since it'll be done inside of a theme:
```
add_theme_support( 'amp', array(
	'template_dir' => 'amp-templates',
```

Reply:
Right, there is the `amp_post_template_dir` filter:

```php
$this->template_dir = apply_filters( 'amp_post_template_dir', AMP__DIR__ . '/templates' );
```

One difference is that the `template_dir` is absolute here, whereas what we're doing in the theme support is relative. I doubt there will be confusion anyway.